src/atomworks/io/utils/io_utils.py

```diff
@@ -337,7 +351,16 @@ def infer_pdb_file_type(path_or_buffer: os.PathLike | io.StringIO | io.BytesIO)
         path_or_buffer.readline()  # Skip the first line
         second_line = path_or_buffer.readline().strip()
         path_or_buffer.seek(0)
-        return "cif" if second_line.startswith("#") else "pdb"
+        path_or_buffer.seek(0)
```

Comment:
The repeated seeks here are a little difficult to read -- maybe we just grab the first and second chars once then pattern match? perhaps something along the lines of:
```
def _infer_filetype(path_or_buffer: io.BytesIO | io.StringIO) -> str:
    """Infer file type from buffer contents."""
    if isinstance(path_or_buffer, io.BytesIO):
        return "bcif"
    
    # StringIO - peek at contents to determine format
    path_or_buffer.seek(0)
    first_char = path_or_buffer.read(1)
    path_or_buffer.readline()  # finish first line
    second_line = path_or_buffer.readline()
    path_or_buffer.seek(0)
    
    if first_char == "{":
        return "mmjson"
    if second_line.startswith("#"):
        return "cif"
    return "pdb"
    ```